.github/workflows/lint.yml

```diff
@@ -1,42 +1,25 @@
-name: Lint
-# Lint runs golangci-lint over the entire cosmos-sdk repository
-# This workflow is run on every pull request and push to main
-# The `golangci` will pass without running if no *.{go, mod, sum} files have been changed.
+name: golangci-lint
 on:
-  pull_request:
   push:
+    tags:
+      - v*
     branches:
       - main
+  pull_request:
 permissions:
   contents: read
-
 jobs:
   golangci:
-    permissions:
-      pull-requests: read  # for technote-space/get-diff-action to get git reference
-    name: golangci-lint
+    name: lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.18
-      - uses: technote-space/get-diff-action@v6.1.0
-        id: git_diff
-        with:
-          PATTERNS: |
-            **/**.go
-            go.mod
-            go.sum
-      - name: Get data from Go build cache
-        # if: env.GIT_DIFF
-        if: ${{ false }}
-        uses: actions/cache@v3
+          go-version: 1.19
```

Comment:
I don't like this approach because it's not what the go tools toolchain uses, i.e. `make lint`. Which means what CI runs can be different than what devs run locally. 

Maybe we investigate why linting doesn't run, perhaps it's a faulty git diff check?

Reply:
Gotta be frank and just say I've seen linters fail 100% of the time -- here and on gaia -- when not using the canonical process. 

What if this was set to latest, and Makefile did like:

```bash
go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```

then they'd both be on latest, and they wouldn't diverge.

100%.  No joke.

Reply:
If you go back to the first commit in this branch, it is what you're suggesting.

Reply:
Yeah I forgot we kinda solved this on gaia:

https://github.com/cosmos/cosmos-sdk/pull/12857